tests/integration/chroot.py

```diff
@@ -0,0 +1,298 @@
+        self._adjust_chroot(tmp_dir)
+        # Start sshd inside the chroot
+        self._start_sshd_chroot(tmp_dir)
+        # Make sure ssh is up and running to avoid
+        # Paramiko error messages
+        self._wait_sshd()
+
+
+    def __del__(self):
+        """ Cleanup resources held by this object """
+        if "keep_running" in self.config and self.config["keep_running"] == True:
+            logger.info("Keeping all resources")
+
+
+    def _validate(self):
+        """ Validate basic settings befor running the tests """
+        # Check if tar archive as image path is defined
+        if not "image" in self.config:
+            logger.error("No path to image archive defined.")
+        else:
+            logger.info("Path to image archive defined: {path}".format(
+              path=self.config["image"]))
+
+        # Check if tar archive as image path  is present
+        image_archive = os.path.exists(self.config["image"])
+        if image_archive:
+            logger.info("Image archive is present: {path}".format(
+              path=self.config["image"]))
+        else:
+            logger.error("Image archive is not present: {path}".format(
+              path=self.config["image"]))
+
+        # Check if port is defined
+        if not "port" in self.config:
+            logger.error("No port for ssh connection defined.")
+        else:
+            logger.info("Port for ssh connection defined: {port}".format(
+              port=self.config["port"]))
+
+        # Check if sshd port can be used
+        self._port_val(self.config["ip"], self.config["port"])
+
+
+    def _unarchive_image(self):
+        """ Unarchive image tar ball """
+        image = self.config["image"]
+        logger.info("Creating tmp_dir")
+        tmp_dir = tempfile.mkdtemp(prefix="gl-int-chroot-")
+        logger.info("Created tmp_dir in {tmp_dir}".format(
+          tmp_dir=tmp_dir))
+        logger.info("Unarchiving image {image} to {tmp_dir}".format(
+          image=image, tmp_dir=tmp_dir))
+        try:
+            with lzma.open(self.config["image"]) as f:
+                with tarfile.open(fileobj=f) as tar:
+                    content = tar.extractall(tmp_dir)
+        except IOError:
+            logger.error("Could not unarchive image file due to IO Error.")
+        except lzma.LZMAError:
+            logger.error("LZMA decompression error.")
+        except tarfile.ReadError:
+            logger.error(("Archive is present but can not be handled by tar."+
+                          "Validate that your tar archive is not corrupt."))
+        except tarfile.CompressionError:
+            logger.error("Used compression is not supported.")
+        except tarfile.HeaderError:
+            logger.error("Malformed tar header information.")
+        logger.info("Unarchived image {image} to {tmp_dir}".format(
+          image=image, tmp_dir=tmp_dir))
+        return tmp_dir
+
+
+    def _generate_ssh_key(self):
```

Comment:
This functionality is needed everywhere. There is a function `RemoteClient.generate_key_pair()` can you reuse that one?

Reply:
Okay, done. But I think `generate_key_pair` should be more fail safe. While playing around with it this function has overwritten my private key. Luckily not a big deal in my case but I think there must be a check to validate if the given file is already present and not to overwrite it by default... If some users want to mount their prod key, this can be very risky...

```
$> md5sum /tmp/ssh_priv_key >> /tmp/foo && pytest --iaas=chroot --configfile=test_config.yaml integration/ && md5sum /tmp/ssh_priv_key >> /tmp/foo && cat /tmp/foo
99a5747c2e338b2a94dc5c58b7a369f9  /tmp/ssh_priv_key
e5f57dd4e229e62e222e095053d7963d  /tmp/ssh_priv_key
```

For `chroot` this isn't a big deal (there's only auto generation available) but it may become one for other cloud platforms. Maybe we just raise a issue for this...